SchedulePlatform.Service/ServiceProvidedService.cs

```diff
@@ -1,55 +1,120 @@
+using SchedulePlatform.Service.Models.ServiceProvided;
 
 namespace SchedulePlatform.Service
 {
     public class ServiceProvidedService : IServiceProvidedService
     {
         private readonly IServiceProvidedRepository _serviceRepository;
-
-        public ServiceProvidedService(IServiceProvidedRepository serviceRepository)
+        private readonly INutritionistServiceRepository _nutritionistServiceRepository;
+        private readonly IMapper _mapper;
+        public ServiceProvidedService(IServiceProvidedRepository serviceRepository, IMapper mapper, INutritionistServiceRepository nutritionistServiceRepository)
         {
             _serviceRepository = serviceRepository;
+            _nutritionistServiceRepository = nutritionistServiceRepository;
+            _mapper = mapper;
         }
 
-        public ServiceProvided[] GetAll()
+        public IEnumerable<ServiceProvidedResponseModel> GetAll()
         {
-            return _serviceRepository.GetAll();
+            var allServices = _serviceRepository.GetAll();
+
+            return _mapper.Map<IEnumerable<ServiceProvidedResponseModel>>(allServices);
         }
 
-        public ServiceProvided Add(ServiceProvided serviceP)
+        public ServiceProvidedResponseModel Add(ServiceProvidedRequestModel serviceP)
         {
-            var serviceToAdd = new ServiceProvided
-            {
-                Id = Guid.NewGuid(),
-                Description = serviceP.Description,
-                NameServiceProvided = serviceP.NameServiceProvided,
-                Price = serviceP.Price,
-                Type = serviceP.Type,
-                UrlPicture = serviceP.UrlPicture
-            };
+            var serviceToAdd = _mapper.Map<ServiceProvided>(serviceP);
 
-            return _serviceRepository.Add(serviceToAdd);
+            _serviceRepository.Add(serviceToAdd);
+
+            return _mapper.Map<ServiceProvidedResponseModel>(serviceToAdd);
         }
 
-        public ServiceProvided? GetById(Guid id)
+        public ServiceProvidedResponseModel GetById(Guid id)
         {
-            return _serviceRepository.GetById(id);
+            var findService = _serviceRepository.GetById(id);
+
+            return _mapper.Map<ServiceProvidedResponseModel>(findService);
         }
 
-        public ServiceProvided Update(ServiceProvided serviceP)
+        public IEnumerable<ServiceProvidedResponseModel> GetAllServicesByNutritionistId(Guid nutritionistId)
         {
-            return _serviceRepository.Update(serviceP);
+            if (_nutritionistServiceRepository.GetAll().ToList().Find(n=>n.NutritionistId==nutritionistId)==null)
+            {
+                throw new Exception("The nutrionist has no related services");
+            }
+
+            var fullList = _nutritionistServiceRepository.GetAll().ToList();
+            var allServices = _nutritionistServiceRepository.GetAll().ToList().Where(n => n.NutritionistId == nutritionistId);
```

Comment:
This call would not be necessary if you store the result of the same call you perform on line 49. If there are services then store them in a list, if not throw exception. 
There is no point in performing the exact same db call twice.